src/main.rs

```diff
@@ -335,6 +336,13 @@ async fn main() -> anyhow::Result<()> {
         policy_changes_topic: args.policy_changes_topic,
         policy_timeout_ms: args.policy_timeout_ms,
     };
+    let labels: HashMap<String, String> = args
+        .label
+        .unwrap_or_default()
+        .iter()
+        .map(|labelpair| parse_label(labelpair))
+        .collect::<Result<HashMap<String, String>, anyhow::Error>>()
```

Comment:
```suggestion
        .collect::<anyhow::Result<HashMap<String, String>>>()
```

Is this type bound really necessary though? I think the compiler should be able to infer this.

Reply:
I thought so too, but it fails without the hint. I think it's because of the `Result`. Originally I had `parse_label` just return the tuple and didn't need the hint